### PR TITLE
feat: [pipe-22104]: add current user to context

### PR DESCRIPTION
**Pull request by cjlee01**
add current user to context

### CODE REVIEW

File: apps/gitness/src/framework/context/AppContext.tsx [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
 import {
   CodeServiceAPIClient,
   TypesMembershipSpace,
   membershipSpaces,
-  TypesSpace
+  TypesSpace,
+  TypesUser,
+  getUser
 } from '@harnessio/code-service-client'
+import { useAtom } from 'jotai'
```
Comment (vardanbansal-harness) [inline apps/gitness/src/framework/context/AppContext.tsx:10]: Lets add a SecOps ticket for this.

**Comment by vardanbansal-harness** [comment]
We haven't implemented cache in this new gitness app yet and I feel we should discuss on a broader level first before we introduce caching as part of this PR. 

For now, if user info is required, let's just add it to the App context **without** the caching part.

File: apps/gitness/src/framework/context/AppContext.tsx (hunk 1 above)
Comment (cjlee01) [reply apps/gitness/src/framework/context/AppContext.tsx:10]: sure will add a secops ticket

File: apps/gitness/src/framework/context/currentUser.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import { TypesUser } from '@harnessio/code-service-client'
```
Comment (vardanbansal-harness) [inline apps/gitness/src/framework/context/currentUser.ts:1]: Lets comment/remove this as well

File: apps/gitness/src/framework/context/AppContext.tsx (hunk 1 above)
Comment (vardanbansal-harness) [reply apps/gitness/src/framework/context/AppContext.tsx:10]: What does `useAtom` do?

Comment (cjlee01) [reply apps/gitness/src/framework/context/AppContext.tsx:10]: Useatom is just a more flexible state management that can update particular states of a object and was used in diffs and PRs and then tan adopted it to other places @vardanbansal-harness

Comment (vardanbansal-harness) [reply apps/gitness/src/framework/context/AppContext.tsx:10]: So it's like a reducer right? @srdjan-harness has already added one here https://github.com/harness/canary/blob/main/apps/gitness/src/hooks/useThunkReducer.tsx

Can we just try to use it? Two different reducers in a single app is not a good idea.

**Review by vardanbansal-harness — APPROVED** [review]

**Pull request merged**